exercises/twelve-days/canonical-data.json

```diff
@@ -0,0 +1,102 @@
+    },
+    {
+      "description": "seventh day seven swans-a-swimming",
+      "property": "verse",
+      "input": 7,
+      "expected": "On the seventh day of Christmas my true love gave to me, seven Swans-a-Swimming, six Geese-a-Laying, five Gold Rings, four Calling Birds, three French Hens, two Turtle Doves, and a Partridge in a Pear Tree.\n"
+    },
+    {
+      "description": "eighth day eight maids-a-milking",
+      "property": "verse",
+      "input": 8,
+      "expected": "On the eighth day of Christmas my true love gave to me, eight Maids-a-Milking, seven Swans-a-Swimming, six Geese-a-Laying, five Gold Rings, four Calling Birds, three French Hens, two Turtle Doves, and a Partridge in a Pear Tree.\n"
+    },
+    {
+      "description": "ninth day nine ladies dancing",
+      "property": "verse",
+      "input": 9,
+      "expected": "On the ninth day of Christmas my true love gave to me, nine Ladies Dancing, eight Maids-a-Milking, seven Swans-a-Swimming, six Geese-a-Laying, five Gold Rings, four Calling Birds, three French Hens, two Turtle Doves, and a Partridge in a Pear Tree.\n"
+    },
+    {
+      "description": "tenth day ten lords-a-leaping",
+      "property": "verse",
+      "input": 10,
+      "expected": "On the tenth day of Christmas my true love gave to me, ten Lords-a-Leaping, nine Ladies Dancing, eight Maids-a-Milking, seven Swans-a-Swimming, six Geese-a-Laying, five Gold Rings, four Calling Birds, three French Hens, two Turtle Doves, and a Partridge in a Pear Tree.\n"
+    },
+    {
+      "description": "eleventh day eleven pipers piping",
+      "property": "verse",
+      "input": 11,
+      "expected": "On the eleventh day of Christmas my true love gave to me, eleven Pipers Piping, ten Lords-a-Leaping, nine Ladies Dancing, eight Maids-a-Milking, seven Swans-a-Swimming, six Geese-a-Laying, five Gold Rings, four Calling Birds, three French Hens, two Turtle Doves, and a Partridge in a Pear Tree.\n"
+    },
+    {
+      "description": "twelfth day twelve drummers drumming",
+      "property": "verse",
+      "input": 12,
+      "expected": "On the twelfth day of Christmas my true love gave to me, twelve Drummers Drumming, eleven Pipers Piping, ten Lords-a-Leaping, nine Ladies Dancing, eight Maids-a-Milking, seven Swans-a-Swimming, six Geese-a-Laying, five Gold Rings, four Calling Birds, three French Hens, two Turtle Doves, and a Partridge in a Pear Tree.\n"
+    },
+    {
+      "description": "finds multiple verses, which are a fragment from the beggining of the song",
+      "property": "verses",
+      "input": [1, 3],
+      "expected": "On the first day of Christmas my true love gave to me, a Partridge in a Pear Tree.\n\nOn the second day of Christmas my true love gave to me, two Turtle Doves, and a Partridge in a Pear Tree.\n\nOn the third day of Christmas my true love gave to me, three French Hens, two Turtle Doves, and a Partridge in a Pear Tree.\n"
```

Comment:
But whenever there are multiple verses they should be separated by `\n\n`
But no trailing newline.

Reply:
But it only makes sense in the context of multiple verses, so why add it to a single verse?

Reply:
> Maybe the trailing single newline should be there.

In this case I am wondering about how many newlines should be at the end of the completed `verses` string. 0, 1 or 2? Currently there is 1.

Reply:
I would expect the following to hold:

1. A single verse would have no newline at the end
1. Verses (more than one) are separated by an empty line
1. Verses don't have trailing newlines